diff -uNrp a/kernel-open/nvidia/os-interface.c b/kernel-open/nvidia/os-interface.c
--- a/kernel-open/nvidia/os-interface.c 2026-09-04 23:33:41.000000000 +0100
+++ b/kernel-open/nvidia/os-interface.c 2026-09-10 06:39:56.650284848 +0100
@@ -1985,7 +1985,7 @@ NV_STATUS NV_API_CALL os_allocate_tempor
 {
     struct file *file = ERR_PTR(-EOPNOTSUPP);
 #if NV_FILESYSTEM_ACCESS_AVAILABLE
-    const char *default_path = "/tmp";
+    const char *default_path = "/var/lib/systemd/sleep";
     const int flags = O_TMPFILE | O_LARGEFILE | O_RDWR;
     const char *path = NVreg_TemporaryFilePath;
 
